plugins/chat/spec/services/actions/create_memberships_for_auto_join_spec.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+RSpec.describe Chat::Action::CreateMembershipsForAutoJoin do
+  subject(:action) { described_class.call(channel: channel, contract: contract) }
+
+  fab!(:channel) { Fabricate(:chat_channel, auto_join_users: true) }
+  fab!(:user_1) { Fabricate(:user, last_seen_at: 15.minutes.ago) }
+
+  let(:start_user_id) { user_1.id }
+  let(:end_user_id) { user_1.id }
+  let(:contract) { OpenStruct.new(start_user_id: start_user_id, end_user_id: end_user_id) }
+
+  it "adds correct members" do
+    expect(action).to eq([user_1.id])
+  end
+
+  it "sets the reason to automatic" do
+    action
+    expect(channel.membership_for(user_1)).to be_automatic
+  end
+
+  context "with others users not in the batch" do
+    fab!(:user_2) { Fabricate(:user) }
+
+    it "adds correct members" do
+      expect(action).to eq([user_1.id])
+    end
+  end
+
+  context "with suspended users" do
+    before { user_1.update!(suspended_till: 1.year.from_now) }
+
+    it "skips suspended users" do
+      expect(action).to eq([])
+    end
+  end
+
+  context "with users not seen recently" do
+    before { user_1.update!(last_seen_at: 4.months.ago) }
+
+    it "skips users last_seen more than 3 months ago" do
+      expect(action).to eq([])
+    end
+  end
+
+  context "with never seen users" do
+    before { user_1.update!(last_seen_at: nil) }
+
+    it "skips users with last_seen set to null" do
+      expect(action).to eq([user_1.id])
```

Comment:
Is it a bug or not in the end? We’re saying it should skip users with `last_seen_at` set to null but that’s not the case it seems :sweat_smile:

Reply:
will check :+1:

Reply:
Yep I checked, the text of the test is wrong (my fault) and it should correctly include this user

Reply:
Perfect then! :partying_face: